dandi/cli/cmd_shell_completion.py

```diff
@@ -0,0 +1,37 @@
+import os
+from os.path import basename
+
+import click
+from packaging.version import Version
+
+SHELLS = ["bash", "zsh", "fish"]
+
+
+@click.command("shell-completion")
+@click.option(
+    "-s",
+    "--shell",
+    type=click.Choice(["auto"] + SHELLS),
+    default="auto",
+    help="The shell for which to generate completion code",
+)
+def shell_completion(shell):
+    """ Emit command completion activation code """
+    if shell == "auto":
```

Comment:
please add the click runner smoke test run so at least not everything in this code block looks red according to coverage

Reply:
What exactly do you mean?

Reply:
via `from click.testing import CliRunner` one -- that it would contribute to coverage as is happening for other `cmd_` click interfaces code like  https://sourcegraph.com/github.com/dandi/dandi-cli/-/blob/dandi/cli/cmd_download.py#L96

Reply:
Due to how Click implements command completion, running the command that way wouldn't work unless the command messes with `sys.argv`, which may not be such a good idea.

Reply:
hm, ok, will trust you on that and let's just proceed